src/transforms/reduce/transform.rs

```diff
@@ -39,13 +52,45 @@ impl ReduceState {
         }
     }
 
-    fn add_event(&mut self, e: LogEvent, strategies: &IndexMap<KeyString, MergeStrategy>) {
+    fn add_event(
+        &mut self,
+        e: LogEvent,
+        strategies: &IndexMap<OwnedTargetPath, MergeStrategy>,
+    ) -> crate::Result<()> {
         self.metadata.merge(e.metadata().clone());
 
+        for (path, strategy) in strategies {
+            if let Some(value) = e.get(path) {
+                match self.fields.entry(path.clone()) {
+                    Entry::Vacant(entry) => match get_value_merger(value.clone(), strategy) {
+                        Ok(m) => {
+                            entry.insert(m);
+                        }
+                        Err(error) => {
+                            warn!(message = "Failed to create value merger.", %error, %path);
+                        }
+                    },
+                    Entry::Occupied(mut entry) => {
+                        if let Err(error) = entry.get_mut().add(value.clone()) {
+                            warn!(message = "Failed to merge value.", %error);
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
I see the warnings here follow the existing pattern for `get_value_merger` failures, but I am wondering if these shouldn't be emitting events that have metrics too 🤔 

I'm ok leaving this for a follow-up since it doesn't seem to make the situation worse.